### PR TITLE
Local Mergin projects are now browser directory items

### LINKED ISSUE #28: Being able to list layers from projects in Mergin
Currently, only the name of projects appear in the Browser panel from plugin. It will be good to be see the list of layers too.

### LINKED ISSUE #236: Mergin project always opened after sync
Sometimes you wish you sync a mergin project but NOT open it.  Nowadays, if there is a sync down or up, when the sync finishes, the mergin project is opened.  When you have another project open already which takes ages to open, this is very annoying.

### PR DESCRIPTION
**Pull request by erpas**
Closes #28
Closes #236 

obraz

### CODE REVIEW

**Comment by erpas** [comment]
It should also close https://github.com/lutraconsulting/qgis-mergin-plugin/issues/236

File: Mergin/plugin.py [hunk 1]
```diff
@@ -298,31 +298,25 @@ def __init__(self, parent, project, project_manager):
         )  # we need posix path for server API calls
         QgsDataItem.__init__(self, QgsDataItem.Collection, parent, self.project_name, "/Mergin/" + self.project_name)
         self.path = mergin_project_local_path(self.project_name)
```
Comment (wonder-sk) [inline Mergin/plugin.py:300]: if it is a remote project, it should have no local path...?

File: Mergin/plugin.py [hunk 2]
```diff
@@ -298,31 +298,25 @@ def __init__(self, parent, project, project_manager):
...
+        self.setSortKey(f"1 {self.name()}")
+        self.setIcon(QIcon(icon_path("cloud-solid.svg")))
         self.project_manager = project_manager
         if self.project_manager is not None:
             self.mc = self.project_manager.mc
         else:
             self.mc = None
 
+    def open_project(self):
+        self.project_manager.open_project(self.path)
```
Comment (wonder-sk) [inline Mergin/plugin.py:310]: if it is a remote project, how can we open the project...?

File: Mergin/plugin.py [hunk 3]
```diff
@@ -369,6 +358,96 @@ def download(self):
...
+        try:
+            self.mc.clone_project(self.project_name, dlg.project_name, dlg.project_namespace)
+            msg = "Mergin project cloned successfully."
+            QMessageBox.information(None, "Clone project", msg, QMessageBox.Close)
+            self.parent().reload()
+            # we also need to reload My projects group as the cloned project could appear there
+            group_items = self.project_manager.get_mergin_browser_groups()
+            if "My projects" in group_items:
+                group_items["My projects"].reload()
+        except (URLError, ClientError) as e:
```
Comment (wonder-sk) [inline Mergin/plugin.py:377]: I would prefer to keep the try/except blocks as short as possible, so it is clear where the error is coming from - e.g. here it should be enough to just wrap the clone_project() call?

**Review by wonder-sk — COMMENTED** [review]
Looking good, just minor bits...

File: Mergin/plugin.py [hunk 4]
```diff
@@ -367,18 +364,18 @@ def clone_remote_project(self):
...
-            self.parent().reload()
-            # we also need to reload My projects group as the cloned project could appear there
-            group_items = self.project_manager.get_mergin_browser_groups()
-            if "My projects" in group_items:
-                group_items["My projects"].reload()
         except (URLError, ClientError) as e:
             msg = "Failed to clone project {}:\n\n{}".format(self.project_name, str(e))
             QMessageBox.critical(None, "Clone project", msg, QMessageBox.Close)
         except LoginError as e:
             login_error_message(e)
```
Comment (wonder-sk) [inline Mergin/plugin.py:371]: let's not forget adding `return` to both exception handlers :-)

**Review by wonder-sk — APPROVED** [review]

**Pull request merged**